node/pkg/node/node.go

```diff
@@ -0,0 +1,574 @@
+				}
+
+				g.acct = accountant.NewAccountant(
+					ctx,
+					logger,
+					g.db,
+					g.obsvReqC.writeC,
+					contract,
+					websocket,
+					g.wormchainConn,
+					enforcing,
+					g.gk,
+					g.gst,
+					g.acctC.writeC,
+					g.env,
+				)
+			} else {
+				logger.Info("acct: accountant is disabled", zap.String("component", "gacct"))
+			}
+
+			return nil
+		}}
+}
+
+func GuardianOptionGovernor(governorEnabled bool) *GuardianOption {
+	return &GuardianOption{
+		name: "governor",
+		f: func(ctx context.Context, logger *zap.Logger, g *G) error {
+			if governorEnabled {
+				logger.Info("chain governor is enabled")
+				g.gov = governor.NewChainGovernor(logger, g.db, g.env)
+			} else {
+				logger.Info("chain governor is disabled")
+			}
+			return nil
+		}}
+}
+
+func GuardianOptionStatusServer(statusAddr string) *GuardianOption {
+	return &GuardianOption{
+		name: "prometheus",
+		f: func(ctx context.Context, logger *zap.Logger, g *G) error {
+			if statusAddr != "" {
+				// Use a custom routing instead of using http.DefaultServeMux directly to avoid accidentally exposing packages
+				// that register themselves with it by default (like pprof).
+				router := mux.NewRouter()
+
+				// pprof server. NOT necessarily safe to expose publicly - only enable it in dev mode to avoid exposing it by
+				// accident. There's benefit to having pprof enabled on production nodes, but we would likely want to expose it
+				// via a dedicated port listening on localhost, or via the admin UNIX socket.
+				if g.env == common.UnsafeDevNet {
+					// Pass requests to http.DefaultServeMux, which pprof automatically registers with as an import side-effect.
+					router.PathPrefix("/debug/pprof/").Handler(http.DefaultServeMux)
+				}
+
+				// Simple endpoint exposing node readiness (safe to expose to untrusted clients)
+				router.HandleFunc("/readyz", readiness.Handler)
+
+				// Prometheus metrics (safe to expose to untrusted clients)
+				router.Handle("/metrics", promhttp.Handler())
+
+				go func() {
+					logger.Info("status server listening", zap.String("status_addr", statusAddr))
+					// SECURITY: If making changes, ensure that we always do `router := mux.NewRouter()` before this to avoid accidentally exposing pprof
+					logger.Error("status server crashed", zap.Error(http.ListenAndServe(statusAddr, router))) // #nosec G114 local status server not vulnerable to DoS attack
+				}()
+			}
+			return nil
+		}}
+}
+
+type IbcWatcherConfig struct {
+	Websocket string
+	Lcd       string
+	Contract  string
+}
+
+func GuardianOptionWatchers(watcherConfigs []watchers.WatcherConfig, ibcWatcherConfig *IbcWatcherConfig) *GuardianOption {
+	return &GuardianOption{
+		name: "watchers",
+		f: func(ctx context.Context, logger *zap.Logger, g *G) error {
+
+			chainObsvReqC := make(map[vaa.ChainID]chan *gossipv1.ObservationRequest)
+
+			go handleReobservationRequests(ctx, clock.New(), logger, g.obsvReqC.readC, chainObsvReqC)
```

Comment:
This appears to be structured differently.  Previously, the watchers were started before handling the reobservation requests.  Is there a reason to change that?

Reply:
The order in which these two goroutines (watcher and reobservation request handler) are started should not matter since they're running concurrently anyways.

Reply:
The code as-is is passing in an empty map to the go func.  I **really** think that the watchers should all be created and the chainObsvReqC map fully filled out before you call this.

Reply:
I think that if you moved this line down to the bottom (just before returning), it would be as before.